packages/mux-uploader/src/mux-uploader-drop.ts

```diff
@@ -0,0 +1,168 @@
+const template = document.createElement('template');
+
+template.innerHTML = `
+<style>
+  /* These styles simulate a user passing these via props. Until
+     that's implemented, we just hardcode it for non-full-screen demo purposes. (TD).
+  */
+  .dropzone {
+    position: absolute;
+    top: 0;
+    left: 0;
+    right: 0;
+    bottom: 0;
+    height: 100%;
+    width: 100%;
+  }
+
+  /* TO-DO: Make default dropzone hover more apparent. (TD).*/
+
+  :host([disableDrop]) .dropzone {
+    display: none;
+  }
+
+  .overlay {
+    display: none;
+  }
+
+  :host([fullScreen]) .dropzone {
+    position: absolute;
+    top: 0;
+    bottom: 0;
+    right: 0;
+    left: 0;
+    height: 100%;
+    width: 100%;
+    z-index: 1;
+  }
+
+  :host([fullScreen][enableOverlay]) .overlay {
+    position: absolute;
+    top: 0;
+    bottom: 0;
+    right: 0;
+    left: 0;
+    height: 100%;
+    width: 100%;
+    z-index: -1;
+  }
+ 
+  :host([active][fullScreen][enableOverlay]) .overlay {
+    z-index: 10;
+    background-color: var(--overlay-background-color, rgba(226, 253, 255, 0.95));
+
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+  }
+
+  h1 {
+    display: none;
+  }
+
+  :host([active][enableOverlay]) h1 {
+    display: block;
+  }
+</style>
+
+<div class="overlay" id="overlay">
+  <h1 id="overlay-text"></h1>
+</div>
+<div class="dropzone" id="dropzone">
+</div>
+`;
+
+class MuxUploaderDrop extends HTMLElement {
+  overlay: HTMLElement | null | undefined;
+  overlayText: HTMLElement | null | undefined;
+
+  constructor() {
+    super();
+    const shadowRoot = this.attachShadow({ mode: 'open' });
+    shadowRoot.appendChild(template.content.cloneNode(true));
+
+    this.overlay = shadowRoot.getElementById('overlay');
+    this.overlayText = shadowRoot.getElementById('overlay-text');
+  }
+
+  connectedCallback() {
+    this.setupDragEvents();
+
+    //@ts-ignore
+    this.shadowRoot.getElementById('overlay-text').innerHTML = this.getAttribute('text');
```

Comment:
nit (blocking): This should be in `attributeChangedCallback` instead of `connectedCallback` so it updates if/when the attribute changes.
nit (non-blocking): Can use `this.overlayText?.innerHTML` that you defined ☝️